specs/feat-phase4-practice-integration.md

```
@@ -0,0 +1,241 @@
+ADD COLUMN section_ids UUID[] DEFAULT NULL;
+
+CREATE INDEX idx_practice_sessions_section_ids
+ON practice_sessions USING GIN (section_ids);
+
+COMMENT ON COLUMN practice_sessions.section_ids IS
+'Array of song_section UUIDs practiced. NULL for legacy sessions. sections_practiced text array kept for display.';
+```
+
+### 6.2 Component Architecture
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)